orchestrator/orchestratorAutomation.ps1

```diff
@@ -35,26 +39,50 @@ Param (
    [string] $storageport
 )
 
-Import-Module 'WebAdministration'
-    if(!$config){   
-        $config = "C:\Program Files (x86)\UiPath\Orchestrator\web.config"
-    }
+Import-Module 'WebAdministration' 
+    
 
-    if(!$aifport){   
-        $aifport = "31390"
-    }
+if(!$config){   
+    $config = "C:\Program Files (x86)\UiPath\Orchestrator"
+} 
 
-    if(!$storageport){
-        $storageport = "31443"
-    }
+#if path does not end with \ add it
+if( $config -notmatch '\\$' ){
+    $config += '\'
+}
 
-    if($domainBasedAccess.Length -gt 0){
-        $domainBasedAccess = $domainBasedAccess.ToString()
-    } else {
-        $domainBasedAccess = "false"
-    }
+$dll_config = $config + 'UiPath.Orchestrator.dll.config'
 
-echo "Path to web.config: "$config
+#Fetching Orchestrator version
+if(Test-Path $dll_config){
+    $orchestrator_version = [System.Diagnostics.FileVersionInfo]::GetVersionInfo($config + 'UiPath.Orchestrator.web.dll').FileVersion
+    echo "Orchestrator version : $orchestrator_version"
+}
+
+if(Test-Path $dll_config){
+    $config = $config + 'UiPath.Orchestrator.dll.config'
+    $configFile = 'UiPath.Orchestrator.dll.config'
+} else{
+    $config = $config + 'web.config'
+    $configFile = 'web.config'
+}
+
+
+if(!$aifport){   
+    $aifport = "31390"
+}
+
+if(!$storageport){
+    $storageport = "31443"
+}
+
+if($domainBasedAccess.Length -gt 0){
+    $domainBasedAccess = $domainBasedAccess.ToString()
+} else {
+    $domainBasedAccess = "false"
+}
+
+echo "Path to Web config: "$config
 
 Copy-Item $config -Destination ("$config.original."+(Get-Date -Format "MMddyyyy.HH.mm.ss"))
 

```

Comment:
i have doubts about this condition as even with domainBasedAccess ... we host aifabric on 31390 only ... please recheck this

Reply:
nope, with domainBasedAccess i.e AKS case, we use DNS name to access ai-app and there is no need of specifying 31390 port over there

Reply:
can we rephrase it to portLess access in that case as for one box installation even with domainBasedAccess, we need port number ?

Reply:
or something better as it was confusing for customer at that point

Reply:
ok, renamed it